DoublyLinkedList/doubly_linked_list.h

```diff
@@ -0,0 +1,255 @@
+#include <iostream>
+using namespace std;
+// Node Creation
+class node
+{
+public:
+    int data;
+    node *next;
+    node *prev;
+};
+
+class doubly_linked_list : private node
+{ //class
+    node *head;
+    node *tail;
+    int length = 0;
+
+public:
+    doubly_linked_list()
+    {
+        head = NULL;
+    };
+    // insertion of value at front
+    void push_front(int value);
+    //insertion of value at end
+    void push_back(int value);
+
+    //for removing values at front and back
+    int remove_front();
+    int remove_back();
+
+    //for inserting at location
+    void insert(int value, int location);
+
+    //for removing at a location
+    void remove_at(int location);
+    void remove(int start, int end);
+
+    //for finding the size
+    size_t size();
+
+    //check whether empty or not
+    bool is_empty();
+
+    //Printing of values
+    void print_values();
+
+    //to clear all the elements in the list
+    void clear();
+
+    int printtail();
+};
+
+void doubly_linked_list::push_front(int value)
+{
+    //new node creation
+    node *new_node = new node();
+    new_node->data = value;
+    new_node->prev = nullptr;
+    new_node->next = (head);
+
+    if (head != nullptr)
+    {
+        head->prev = new_node; //if head node is not empty then prev of head node will be assigned to the new node
+    }
+    if (head == nullptr)
+    {
+        tail = new_node;
+    }
+    length++;
+    head = new_node; //now the new node becomes the first node
+}
+void doubly_linked_list::push_back(int value) //add the element at the back
+{
+    //new node creation
+    node *new_node = new node();
+    new_node->data = value;
+    new_node->prev = tail;
+    new_node->next = nullptr;
+
+    if (tail != nullptr)
+    {
+        tail->next = new_node; //if tail node is not empty then prev of head node will be assigned to the new node
+    }
+    if (head == tail)
+    {
+        head = new_node;
+    }
+    length++;
+    tail = new_node; //now the new node becomes the last node
+}
+
+int doubly_linked_list::remove_front()
+{
+    int temp = head->data;
+    //assigning the prev of second node as NULL because it becomes the first node
+    head->next->prev = nullptr;
+    //new node will point to the head node
+    head = head->next;
+    //now the new node becomes the first node
+    length--;
+    return temp;
+}
+int doubly_linked_list::remove_back()
+{
+    //for finding the last node
+    int temp = tail->data;
+    tail->prev->next = nullptr;
+    length--;
+    tail = tail->prev;
+    return temp;
+}
+
+void doubly_linked_list::insert(int value, int location) //for inserting at a location
+{
+    node *new_node = new node(); //new doubly_linked_list creation
```

Comment:
what happens if location is negative? 

what happens if location is greater than length ?